Classes/DownloadHandler.cs

```diff
@@ -190,7 +190,7 @@ private void DownloadProgThread()
 
                 try
                 {
-                    saveLocation = FileUtils.GetSaveFolder();
+                    saveLocation = FileUtils.GetSaveFolder(this.progInfo);
```

Comment:
Then you have to use this.progInfo.Progid all the time you call such a function. Beside that one might at some point decide, in principle, to take another property of progInfo to use. Now it's the responsibility of only Get/SetValuePostfix. It might even be that the settings become part of progInfo. However, in the Main Options GUI progInfo might not be naturally available.

The first version I made was at the other end of responsibility, it took a string, which it put after the setting name.